fe/fe-core/src/main/java/com/starrocks/connector/iceberg/CachingIcebergCatalog.java

```diff
@@ -0,0 +1,149 @@
+import com.starrocks.connector.statistics.ConnectorColumnStatisticLoader;
+import com.starrocks.connector.statistics.ExternalTableColumnKey;
+import com.starrocks.connector.statistics.ExternalTableColumnStat;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+public class CachingIcebergCatalog implements IcebergCatalog {
+    private static final Logger LOG = LogManager.getLogger(CachingIcebergCatalog.class);
+    private IcebergCatalog delegate;
+    private final AsyncLoadingCache<ExternalTableColumnKey, Optional<ExternalTableColumnStat>> cacheTableStats;
+
+    public CachingIcebergCatalog(IcebergCatalog icebergCatalog) {
+        this.delegate = icebergCatalog;
+        cacheTableStats = Caffeine.newBuilder().maximumSize(Config.iceberg_meta_cache_size)
+                .expireAfterAccess(Config.iceberg_meta_cache_ttl_s, TimeUnit.SECONDS)
+                .buildAsync(new ConnectorColumnStatisticLoader());
+    }
+
+    @Override
+    public IcebergCatalogType getIcebergCatalogType() {
+        return delegate.getIcebergCatalogType();
+    }
+
+    @Override
+    public List<String> listAllDatabases() {
+        return delegate.listAllDatabases();
+    }
+
+    @Override
+    public void createDb(String dbName, Map<String, String> properties) {
+        delegate.createDb(dbName, properties);
+    }
+
+    @Override
+    public void dropDb(String dbName) throws MetaNotFoundException {
+        delegate.dropDb(dbName);
+    }
+
+    @Override
+    public Database getDB(String dbName) {
+        return delegate.getDB(dbName);
+    }
+
+    @Override
+    public List<String> listTables(String dbName) {
+        return delegate.listTables(dbName);
+    }
+
+    @Override
+    public boolean createTable(String dbName, String tableName, Schema schema, PartitionSpec partitionSpec,
+                               String location, Map<String, String> properties) {
+        return delegate.createTable(dbName, tableName, schema, partitionSpec, location, properties);
+    }
+
+    @Override
+    public boolean dropTable(String dbName, String tableName, boolean purge) {
+        return delegate.dropTable(dbName, tableName, purge);
+    }
+
+    @Override
+    public Table getTable(String dbName, String tableName) throws StarRocksConnectorException {
+        return delegate.getTable(dbName, tableName);
+    }
+
+    @Override
+    public void deleteUncommittedDataFiles(List<String> fileLocations) {
+        delegate.deleteUncommittedDataFiles(fileLocations);
+    }
+
+    public List<Optional<ExternalTableColumnStat>> getColumnStatistics(String tableUUID, List<String> columns) {
```

Comment:
What tableUUID looks like?

Reply:
for native table it's tableID->string, for external table it's catalogName.dbName.tableName.(create time or native UUID)